internal/services/networkfunction/network_function_collector_policy_resource.go

```diff
@@ -0,0 +1,495 @@
+	EmissionType         collectorpolicies.EmissionType   `tfschema:"emission_type"`
+}
+
+type EmissionPolicyDestinationModel struct {
+	DestinationType collectorpolicies.DestinationType `tfschema:"destination_type"`
+}
+
+type IngestionPolicyPropertiesFormatModel struct {
+	IngestionSources []IngestionSourcesPropertiesFormatModel `tfschema:"ingestion_source"`
+	IngestionType    collectorpolicies.IngestionType         `tfschema:"ingestion_type"`
+}
+
+type IngestionSourcesPropertiesFormatModel struct {
+	ResourceId string                       `tfschema:"resource_id"`
+	SourceType collectorpolicies.SourceType `tfschema:"source_type"`
+}
+
+type NetworkFunctionCollectorPolicyResource struct{}
+
+var _ sdk.ResourceWithUpdate = NetworkFunctionCollectorPolicyResource{}
+
+func (r NetworkFunctionCollectorPolicyResource) ResourceType() string {
+	return "azurerm_network_function_collector_policy"
+}
+
+func (r NetworkFunctionCollectorPolicyResource) ModelObject() interface{} {
+	return &NetworkFunctionCollectorPolicyModel{}
+}
+
+func (r NetworkFunctionCollectorPolicyResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return collectorpolicies.ValidateCollectorPolicyID
+}
+
+func (r NetworkFunctionCollectorPolicyResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[a-zA-Z0-9]([-._a-zA-Z0-9]{0,78}[a-zA-Z0-9_])?$"),
+				"The name can contain only letters, numbers, periods (.), hyphens (-),and underscores (_), up to 80 characters, and it must begin with a letter or number and end with a letter, number or underscore.",
+			),
+		},
+
+		"location": commonschema.Location(),
+
+		"traffic_collector_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: azuretrafficcollectors.ValidateAzureTrafficCollectorID,
+		},
+
+		"emission_policy": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"emission_destination": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MinItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"destination_type": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ForceNew: true,
+									ValidateFunc: validation.StringInSlice([]string{
+										string(collectorpolicies.DestinationTypeAzureMonitor),
+									}, false),
+								},
+							},
+						},
+					},
+
+					"emission_type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(collectorpolicies.EmissionTypeIPFIX),
+						}, false),
+					},
```

Comment:
I'm unsure about the design of this - we can have multiple `emission_policy` and `emission_destinations` but all only of the same type since only one value is accepted. What does it mean to supply multiple of these when they're identical?

Could we potentially flatten `emission_destination` and `destination_type` into one property?

Reply:
For `emission_policy`, the service team said that they will add more enum values for these properties. So, I still suggest keeping current definition. Or, if new enum values are added in future, it will be hard for us to update the schema without breaking change in the same Major version. I added `MaxItems: 1` to limit the item numbers and can be removed when new enum values are added in future.

For `ingestion_policy`, since there is only 1 elemnt, I have flattened the inner properties.